### PR TITLE
Add new pprof.LogTrace RPC and make UI use it

### PR DESCRIPTION
**Pull request by akalin-keybase**
Have it dump trace files into the log directory and keep the five
most recent traces.

Also clean up AppBundleForPath a bit.

### CODE REVIEW

**Comment by akalin-keybase** [comment]
@mlsteele for go side, @keybase/react-hackers for JS side

**Comment by mlsteele** [comment]
Why do we want to upload these with log send? Can they contain secrets that we wouldn't otherwise send with log send?

**Comment by akalin-keybase** [comment]
It's so that we get profiling info from production builds, to debug e.g. memory issues.

From reading the trace format docs ( https://docs.google.com/document/d/1FP5apqzBgr7ahCCgFO-yoVhk4YZrNIDNf9RybngBc14/edit ) I believe it won't contain any secrets.

File: go/client/cmd_pprof.go [hunk 1]
```diff
@@ -42,17 +54,25 @@ func (c *CmdPprofTrace) Run() error {
 	if err = RegisterProtocolsWithContext(nil, c.G()); err != nil {
 		return err
 	}
+
+	durationSeconds := keybase1.DurationSec(float64(c.traceDuration) / float64(time.Second))
```
Comment (mlsteele) [inline go/client/cmd_pprof.go]: c.traceDuration.Seconds()

File: go/client/cmd_pprof.go [hunk 2]
```diff
@@ -63,6 +83,10 @@ func NewCmdPprofTrace(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Co
 				Value: 5 * time.Second,
 				Usage: "How long to run the trace.",
 			},
+			cli.BoolFlag{
+				Name:  "log, l",
+				Usage: "Whether to write the trace to the log directory. If set, don't pass an argument.",
```
Comment (mlsteele) [inline go/client/cmd_pprof.go]: Mention here that it gets log sent

**Comment by akalin-keybase** [comment]
The default behavior for `keybase pprof trace` is to take a path, and it's only when `-l` is specified that it puts it in the log directory (and will eventually include it in a log send). Changed usage to mention that explicitly.

The button in the advanced section of the UI also puts it in the log directory, so added a blurb for that. This is primarily for mobile, where it isn't easy to get files off of the phone, especially for non-developers, which is why I want to include it in a log send.

**Pull request merged**